indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,419 @@
+        const {roomAttributes} = this.state;
+        this.setState({roomAttributes: _.omit(roomAttributes, attribute)});
+    };
+
+    renderAttributes = () => {
+        const {attributes, roomAttributes} = this.state;
+        if (!attributes) {
+            return;
+        }
+        return attributes.map(attribute => (
+            <>
+                {_.has(roomAttributes, attribute.name) && (
+                    <Field name={attribute.name}
+                           component={ReduxFormField}
+                           label={attribute.name}
+                           as={Input}
+                           icon={{name: 'remove', color: 'red', circular: true, link: true, onClick: () => this.removeAttribute(attribute.name)}} />
+                )} </>
+
+        ));
+    };
+
+    renderAddButton = (type) => {
+        const {attributes} = this.state;
+        return (
+            <Button disabled={attributes.length === 0}
+                    default
+                    size="tiny"
+                    floated="right"
+                    onClick={type === 'attributes' ? this.onEditAttributes : this.onEditHours}>
+                Add
+            </Button>
+        );
+    };
+
+    renderColumn = (column, key) => (
+        <Grid.Column key={key}>{column.map(this.renderContent)}</Grid.Column>
+    );
+
+    renderContent = (content) => {
+        const {room, attributes} = this.state;
+        let options;
+        switch (content.type) {
+            case 'header':
+                return (
+                    <Header key={content.label}>{content.label}</Header>);
+            case 'input':
+                return (
+                    <Field key={content.name}
+                           name={content.name}
+                           component={ReduxFormField}
+                           label={content.label}
+                           as="input" />);
+            case 'formgroup':
+                return (
+                    <Form.Group key={content.key}>
+                        {content.content.map(this.renderContent)}
+                    </Form.Group>);
+            case 'checkbox':
+                return (
+                    <Field key={content.name}
+                           name={content.name}
+                           component={ReduxCheckboxField}
+                           componentLabel={content.label}
+                           as={Checkbox} />);
+            case 'image':
+                return this.renderImage(room.spritePosition);
+            case 'attributes':
+                return this.renderAttributes();
+            case 'addButton':
+                return this.renderAddButton(content.editType);
+            case 'dropdown':
+                if (!attributes) {
+                    return;
+                }
+                options = this.generateOptions(attributes);
+                return (
+                    <Dropdown placeholder="Add new attributes"
```

Comment:
ok, "type dropdown" is not the right choice for this entry. I think in terms of the declarative list on top, there should be one entry for "attributes" and this the render function for this should render *everything* needed for them - i.e. both the dropdown and the fields for the currently-set attributes